### PR TITLE
Fix default extinction correction (again)

### PR DESCRIPTION
**Pull request by finagle29**
#1215 didn't actually fix the default extinction correction because the sensfunc algorithm wasn't being read from sens.fits because algorithm was not a part of the datamodel.

This PR adds the sensfunc algorithm to the sensfunc datamodel, which finally fixes the default extinction correction.

### CODE REVIEW

File: pypeit/sensfunc.py [hunk 1]
```diff
@@ -170,6 +171,9 @@ def __init__(self, spec1dfile, sensfile, par=None, debug=False):
         self.par = self.spectrograph.default_pypeit_par()['sensfunc'] if par is None else par
         # TODO: Check the type of the parameter object?
 
+        # Set the algorithm in the datamodel
+        self.algorithm = self.__class__.algorithm
```
Comment (kbwestfall) [inline pypeit/sensfunc.py]: I'm worried that naming the instance attribute the same as the class attribute might come back to bite us.  Can we instead rename the class attribute to `_algorithm` or something?

File: pypeit/sensfunc.py [hunk 2]
```diff
@@ -844,6 +845,7 @@ def __init__(self, spec1dfile, sensfile, par=None, debug=False):
         # already in unpack object
         self.meta_spec['LATITUDE'] = self.spectrograph.telescope['latitude']
         self.meta_spec['LONGITUDE'] = self.spectrograph.telescope['longitude']
+        #self.algorithm = self.__class__.algorithm
```
Comment (kbwestfall) [inline pypeit/sensfunc.py]: remove?

**Review by kbwestfall — CHANGES_REQUESTED** [review]
Good catch!  Was the root of the problem the fact that the sensfunc file is being read by the base class (`SensFunc`) and not the relevant subclass (e.g., `IRSensFunc`)?  If so, it would be good to understand if we can address the root of the problem, but this feels like a reasonable compromise.

I have a request on a name change, and I think tests are failing because you need to update the `sens*` file that's kept in the test directory.

**Comment by finagle29** [comment]
Yeah I believe the root cause is that the file is being read in by the base class `SensFunc` instead of the relevant subclass.

File: pypeit/sensfunc.py (hunk 1 above)
Comment (finagle29) [reply pypeit/sensfunc.py]: yeah, sounds good

File: pypeit/sensfunc.py (hunk 2 above)
Comment (finagle29) [reply pypeit/sensfunc.py]: done

**Review by profxj — APPROVED** [review]
approving once Kyle is happy.

**Comment by finagle29** [comment]
@kbwestfall I think I need to update `pypeit/tests/files/sens_cN20170331S0206-HIP62745_GNIRS_20170331T083351.681.fits` to the new datamodel version, can you let me know how you updated that for your sensfunc datamodel PR?

**Comment by profxj** [comment]
Actually, one potentially challenging request from me:

can we add a Test (unit or DevSuite) so we *know* this 
is being applied?

**Review by profxj — CHANGES_REQUESTED** [review]
made a test request

**Review by jhennawi — APPROVED** [review]
Thanks for catching this. Please test it for both IR and UVIS examples in the dev suite.

**Comment by kbwestfall** [comment]
> @kbwestfall I think I need to update `pypeit/tests/files/sens_cN20170331S0206-HIP62745_GNIRS_20170331T083351.681.fits` to the new datamodel version, can you let me know how you updated that for your sensfunc datamodel PR?

If I remember correctly, the easiest way to get this file is to run the Dev-Suite on the gemini_gnirs/32_SB_SXD data.

**Review by kbwestfall — APPROVED** [review]
Excellent!  Thanks for the new tests!

**Review by profxj — APPROVED** [review]

**Comment by finagle29** [comment]
merging

**Pull request merged**